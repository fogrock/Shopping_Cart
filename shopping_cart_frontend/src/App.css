.mainButton {
    margin-top: 20px;
    float: right
}

.cartPostage {
    padding-left: 880px;
    font-family: Calibri;
}

.cartTotals {
    padding-left: 880px;
    font-family: Calibri;
    font-weight: 700;
}

.paddedUpDown {
    
    margin-top: 10px;
    margin-bottom: 10px;
}