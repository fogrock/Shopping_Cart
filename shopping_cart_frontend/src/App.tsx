import './App.css';
import React, { useEffect } from 'react';
import Alert from '@mui/material/Alert';
import { ApiProxy } from './shopping/ApiProxy';
import InputLabel from '@mui/material/InputLabel';
import Select, { SelectChangeEvent } from '@mui/material/Select';
import MenuItem from '@mui/material/MenuItem';
import FormControl from '@mui/material/FormControl';
import { Products } from './shopping/Products';
import { CartView } from './shopping/CartView';
import Button from '@mui/material/Button';
import { ThankYou } from './shopping/ThankYou';

function App() {
    const [error, setError] = React.useState<string | undefined>(undefined);
    const [countries, setCountries] = React.useState<string[]>(['Australia']);
    const [countryName, setCountryName] = React.useState<string>('Australia');
    const [showProducts, setShowProducts] = React.useState<boolean>(true);
    const [showCart, setShowCart] = React.useState<boolean>(false);
    const [showThankYou, setShowThankYou] = React.useState<boolean>(false);

    useEffect(() => {
        getCountryList();
    }, []);

    const getCountryList = () => {
        const apiProxy = new ApiProxy();
        apiProxy.getCountryList()
            .then(result => setCountries(result))
            .catch(error => setError(error.response.data))
    }

    const onCountryChange = (event: SelectChangeEvent<typeof countryName>) => {
        const { target: { value } } = event;
        setCountryName(value);
    };

    const onGoToCart = () => {
        setShowProducts(false);
        setShowCart(true);
        setShowThankYou(false)
    }

    const onGoToProduct = () => {
        setShowProducts(true);
        setShowCart(false);
        setShowThankYou(false)
    }

    const onOrderProcessed = () => {
        setShowProducts(false);
        setShowCart(false);
        setShowThankYou(true)
    }

    return (
        <div>
            {error ? <div className="paddedUpDown"><Alert variant='outlined' severity='error'>Error: {error}</Alert></div> : null}
            <FormControl variant="standard" sx={{ m: 1, minWidth: 120 }}>
                <InputLabel>Country</InputLabel>
                <Select
                    style={{ width: '200px', marginBottom: '10px' }}
                    labelId="country-label"
                    id="country-select"
                    value={countryName}
                    label="Country"
                    onChange={onCountryChange}
                >
                    {countries.map((country) => (
                        <MenuItem key={country} value={country}>{country}</MenuItem>
                    ))}
                </Select>
            </FormControl>
            {showProducts ? <Button variant="contained" onClick={onGoToCart} className='mainButton'>Go to shopping cart</Button> : null}
            {showCart || showThankYou ? <Button variant="contained" onClick={onGoToProduct} className='mainButton'>Go to product list</Button> : null}
            {showProducts ? <Products countryName={countryName} /> : null}
            {showCart ? <CartView countryName={countryName} onOrderProcessed={onOrderProcessed} /> : null}
            {showThankYou ? <ThankYou /> : null}
        </div>
    );
}

export default App;
