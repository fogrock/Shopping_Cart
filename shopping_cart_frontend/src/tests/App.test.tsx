import React from 'react';
import { render, screen } from '@testing-library/react';
import App from '../App';

test('renders buttons and selector', () => {
    render(<App />);
    const linkElement1 = screen.getByText(/Go to shopping cart/i);
    expect(linkElement1).toBeInTheDocument();
    const linkElement2 = screen.getByText(/Country/i);
    expect(linkElement2).toBeInTheDocument();
});

it("matches snapshot", () => {
    const { asFragment } = render(<App />);
    expect(asFragment()).toMatchSnapshot();
});