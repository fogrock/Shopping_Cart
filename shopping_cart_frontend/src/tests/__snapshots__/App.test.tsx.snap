// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiFormControl-root css-1869usk-MuiFormControl-root"
    >
      <label
        class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-standard MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1c2i806-MuiFormLabel-root-MuiInputLabel-root"
        data-shrink="true"
      >
        Country
      </label>
      <div
        class="MuiInput-root MuiInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-a3l6o-MuiInputBase-root-MuiInput-root-MuiSelect-root"
        style="width: 200px; margin-bottom: 10px;"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-labelledby="country-label country-select"
          class="MuiSelect-select MuiSelect-standard MuiInput-input MuiInputBase-input css-1rxz5jq-MuiSelect-select-MuiInputBase-input-MuiInput-input"
          id="country-select"
          role="button"
          tabindex="0"
        >
          Australia
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="Australia"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconStandard css-pqjvzy-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ArrowDropDownIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
    <button
      class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root mainButton css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Go to shopping cart
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <div>
      <div
        class="ag-theme-alpine paddedUpDown"
      >
        <div
          style="height: 100%;"
        >
          <!-- AG Grid -->
          <div
            class="ag-root-wrapper ag-ltr ag-layout-auto-height"
          >
            <div
              class="ag-root-wrapper-body ag-focus-managed ag-layout-auto-height"
            >
              <div
                class="ag-tab-guard ag-tab-guard-top"
                role="presentation"
                tabindex="0"
              />
              <!-- AG Grid Body -->
              <div
                aria-colcount="4"
                aria-rowcount="1"
                class="ag-root ag-unselectable ag-layout-auto-height"
                role="grid"
              >
                <div
                  class="ag-header ag-pivot-off"
                  role="presentation"
                  style="height: 49px; min-height: 49px;"
                >
                  <div
                    class="ag-pinned-left-header ag-hidden"
                    role="presentation"
                    style="width: 0px; min-width: 0px; max-width: 0px;"
                  >
                    <div
                      aria-rowindex="1"
                      class="ag-header-row ag-header-row-column"
                      role="row"
                      style="height: 48px; top: 0px; width: 0px;"
                    />
                  </div>
                  <div
                    class="ag-header-viewport "
                    role="presentation"
                  >
                    <div
                      class="ag-header-container"
                      role="rowgroup"
                      style="width: 1230px;"
                    >
                      <div
                        aria-rowindex="1"
                        class="ag-header-row ag-header-row-column"
                        role="row"
                        style="height: 48px; top: 0px; width: 1230px;"
                      >
                        <div
                          aria-colindex="1"
                          aria-description="Press ENTER to sort."
                          aria-sort="none"
                          class="ag-header-cell ag-header-cell-sortable ag-focus-managed"
                          col-id="0"
                          role="columnheader"
                          style="left: 0px; width: 50px;"
                          tabindex="-1"
                        >
                          <div
                            class="ag-header-cell-resize ag-hidden"
                            role="presentation"
                          />
                          <div
                            class="ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-header-select-all ag-hidden"
                            role="presentation"
                          >
                            
                
                            <div
                              class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                              ref="eLabel"
                              role="presentation"
                            />
                            
                
                            <div
                              class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                              ref="eWrapper"
                              role="presentation"
                            >
                              
                    
                              <input
                                class="ag-input-field-input ag-checkbox-input"
                                id="ag-16-input"
                                ref="eInput"
                                tabindex="-1"
                                type="checkbox"
                              />
                              
                
                            </div>
                            
            
                          </div>
                          <div
                            class="ag-cell-label-container ag-header-cell-sorted-none"
                            role="presentation"
                          >
                            
            
            
                            <div
                              class="ag-header-cell-label"
                              ref="eLabel"
                              role="presentation"
                            >
                              
                
                              <span
                                class="ag-header-cell-text"
                                ref="eText"
                              />
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                                ref="eFilter"
                              >
                                <span
                                  class="ag-icon ag-icon-filter"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                                ref="eSortOrder"
                              />
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                                ref="eSortAsc"
                              >
                                <span
                                  class="ag-icon ag-icon-asc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                                ref="eSortDesc"
                              >
                                <span
                                  class="ag-icon ag-icon-desc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                                ref="eSortNone"
                              >
                                <span
                                  class="ag-icon ag-icon-none"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
            
                            </div>
                            
        
                          </div>
                        </div>
                        <div
                          aria-colindex="2"
                          aria-description="Press ENTER to sort."
                          aria-sort="none"
                          class="ag-header-cell ag-header-cell-sortable ag-focus-managed"
                          col-id="productName"
                          role="columnheader"
                          style="left: 50px; width: 400px;"
                          tabindex="-1"
                        >
                          <div
                            class="ag-header-cell-resize ag-hidden"
                            role="presentation"
                          />
                          <div
                            class="ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-header-select-all ag-hidden"
                            role="presentation"
                          >
                            
                
                            <div
                              class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                              ref="eLabel"
                              role="presentation"
                            />
                            
                
                            <div
                              class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                              ref="eWrapper"
                              role="presentation"
                            >
                              
                    
                              <input
                                class="ag-input-field-input ag-checkbox-input"
                                id="ag-17-input"
                                ref="eInput"
                                tabindex="-1"
                                type="checkbox"
                              />
                              
                
                            </div>
                            
            
                          </div>
                          <div
                            class="ag-cell-label-container ag-header-cell-sorted-none"
                            role="presentation"
                          >
                            
            
            
                            <div
                              class="ag-header-cell-label"
                              ref="eLabel"
                              role="presentation"
                            >
                              
                
                              <span
                                class="ag-header-cell-text"
                                ref="eText"
                              >
                                Name
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                                ref="eFilter"
                              >
                                <span
                                  class="ag-icon ag-icon-filter"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                                ref="eSortOrder"
                              />
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                                ref="eSortAsc"
                              >
                                <span
                                  class="ag-icon ag-icon-asc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                                ref="eSortDesc"
                              >
                                <span
                                  class="ag-icon ag-icon-desc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                                ref="eSortNone"
                              >
                                <span
                                  class="ag-icon ag-icon-none"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
            
                            </div>
                            
        
                          </div>
                        </div>
                        <div
                          aria-colindex="3"
                          aria-description="Press ENTER to sort."
                          aria-sort="none"
                          class="ag-header-cell ag-header-cell-sortable ag-focus-managed"
                          col-id="productDescription"
                          role="columnheader"
                          style="left: 450px; width: 600px;"
                          tabindex="-1"
                        >
                          <div
                            class="ag-header-cell-resize"
                            role="presentation"
                          />
                          <div
                            class="ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-header-select-all ag-hidden"
                            role="presentation"
                          >
                            
                
                            <div
                              class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                              ref="eLabel"
                              role="presentation"
                            />
                            
                
                            <div
                              class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                              ref="eWrapper"
                              role="presentation"
                            >
                              
                    
                              <input
                                class="ag-input-field-input ag-checkbox-input"
                                id="ag-18-input"
                                ref="eInput"
                                tabindex="-1"
                                type="checkbox"
                              />
                              
                
                            </div>
                            
            
                          </div>
                          <div
                            class="ag-cell-label-container ag-header-cell-sorted-none"
                            role="presentation"
                          >
                            
            
            
                            <div
                              class="ag-header-cell-label"
                              ref="eLabel"
                              role="presentation"
                            >
                              
                
                              <span
                                class="ag-header-cell-text"
                                ref="eText"
                              >
                                Description
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                                ref="eFilter"
                              >
                                <span
                                  class="ag-icon ag-icon-filter"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                                ref="eSortOrder"
                              />
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                                ref="eSortAsc"
                              >
                                <span
                                  class="ag-icon ag-icon-asc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                                ref="eSortDesc"
                              >
                                <span
                                  class="ag-icon ag-icon-desc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                                ref="eSortNone"
                              >
                                <span
                                  class="ag-icon ag-icon-none"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
            
                            </div>
                            
        
                          </div>
                        </div>
                        <div
                          aria-colindex="4"
                          aria-description="Press ENTER to sort."
                          aria-sort="none"
                          class="ag-header-cell ag-header-cell-sortable ag-focus-managed"
                          col-id="price.amount"
                          role="columnheader"
                          style="left: 1050px; width: 180px;"
                          tabindex="-1"
                        >
                          <div
                            class="ag-header-cell-resize ag-hidden"
                            role="presentation"
                          />
                          <div
                            class="ag-labeled ag-label-align-right ag-checkbox ag-input-field ag-header-select-all ag-hidden"
                            role="presentation"
                          >
                            
                
                            <div
                              class="ag-input-field-label ag-label ag-hidden ag-checkbox-label"
                              ref="eLabel"
                              role="presentation"
                            />
                            
                
                            <div
                              class="ag-wrapper ag-input-wrapper ag-checkbox-input-wrapper"
                              ref="eWrapper"
                              role="presentation"
                            >
                              
                    
                              <input
                                class="ag-input-field-input ag-checkbox-input"
                                id="ag-19-input"
                                ref="eInput"
                                tabindex="-1"
                                type="checkbox"
                              />
                              
                
                            </div>
                            
            
                          </div>
                          <div
                            class="ag-cell-label-container ag-header-cell-sorted-none"
                            role="presentation"
                          >
                            
            
            
                            <div
                              class="ag-header-cell-label"
                              ref="eLabel"
                              role="presentation"
                            >
                              
                
                              <span
                                class="ag-header-cell-text"
                                ref="eText"
                              >
                                Price
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-filter-icon ag-hidden"
                                ref="eFilter"
                              >
                                <span
                                  class="ag-icon ag-icon-filter"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-order ag-hidden"
                                ref="eSortOrder"
                              />
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-ascending-icon ag-hidden"
                                ref="eSortAsc"
                              >
                                <span
                                  class="ag-icon ag-icon-asc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-descending-icon ag-hidden"
                                ref="eSortDesc"
                              >
                                <span
                                  class="ag-icon ag-icon-desc"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
                
                              <span
                                aria-hidden="true"
                                class="ag-header-icon ag-header-label-icon ag-sort-none-icon ag-hidden"
                                ref="eSortNone"
                              >
                                <span
                                  class="ag-icon ag-icon-none"
                                  role="presentation"
                                  unselectable="on"
                                />
                              </span>
                              
            
                            </div>
                            
        
                          </div>
                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    class="ag-pinned-right-header ag-hidden"
                    role="presentation"
                    style="width: 0px; min-width: 0px; max-width: 0px;"
                  >
                    <div
                      aria-rowindex="1"
                      class="ag-header-row ag-header-row-column"
                      role="row"
                      style="height: 48px; top: 0px; width: 0px;"
                    />
                  </div>
                </div>
                <!-- AG Pinned Top -->
                <div
                  class="ag-floating-top"
                  role="presentation"
                  style="height: 0px; min-height: 0; display: none; overflow-y: hidden;"
                >
                  <!-- AG Row Container topLeft -->
                  <div
                    class="ag-pinned-left-floating-top ag-hidden"
                    role="presentation"
                  />
                  <!-- AG Row Container topCenter -->
                  <div
                    class="ag-floating-top-viewport"
                    role="presentation"
                  >
                    <div
                      class="ag-floating-top-container"
                      role="presentation"
                      style="width: 1230px;"
                    />
                  </div>
                  <!-- AG Row Container topRight -->
                  <div
                    class="ag-pinned-right-floating-top ag-hidden"
                    role="presentation"
                  />
                  <!-- AG Row Container topFullWidth -->
                  <div
                    class="ag-floating-top-full-width-container"
                    role="presentation"
                  />
                </div>
                <!-- AG Middle -->
                <div
                  class="ag-body-viewport ag-row-no-animation ag-layout-auto-height"
                  role="presentation"
                >
                  <!-- AG Row Container left -->
                  <div
                    class="ag-pinned-left-cols-container ag-hidden"
                    role="presentation"
                    style="height: 1px;"
                  />
                  <!-- AG Row Container center -->
                  <div
                    class="ag-center-cols-clipper"
                    role="presentation"
                    style="height: 1px;"
                  >
                    <div
                      class="ag-center-cols-viewport"
                      role="presentation"
                      style="height: 100%;"
                    >
                      <div
                        class="ag-center-cols-container"
                        role="presentation"
                        style="height: 1px; width: 1230px;"
                      />
                    </div>
                  </div>
                  <!-- AG Row Container right -->
                  <div
                    class="ag-pinned-right-cols-container ag-hidden"
                    role="presentation"
                    style="height: 1px;"
                  />
                  <!-- AG Row Container fullWidth -->
                  <div
                    class="ag-full-width-container"
                    role="presentation"
                    style="height: 1px;"
                  />
                </div>
                <!-- AG Pinned Bottom -->
                <div
                  class="ag-floating-bottom"
                  role="presentation"
                  style="height: 0px; min-height: 0; display: none; overflow-y: hidden;"
                >
                  <!-- AG Row Container bottomLeft -->
                  <div
                    class="ag-pinned-left-floating-bottom ag-hidden"
                    role="presentation"
                  />
                  <!-- AG Row Container bottomCenter -->
                  <div
                    class="ag-floating-bottom-viewport"
                    role="presentation"
                  >
                    <div
                      class="ag-floating-bottom-container"
                      role="presentation"
                      style="width: 1230px;"
                    />
                  </div>
                  <!-- AG Row Container bottomRight -->
                  <div
                    class="ag-pinned-right-floating-bottom ag-hidden"
                    role="presentation"
                  />
                  <!-- AG Row Container bottomFullWidth -->
                  <div
                    class="ag-floating-bottom-full-width-container"
                    role="presentation"
                  />
                </div>
                <!-- AG Fake Horizontal Scroll -->
                <div
                  aria-hidden="true"
                  class="ag-body-horizontal-scroll"
                  style="height: 0px; max-height: 0px; min-height: 0px;"
                >
                  
            
                  <div
                    class="ag-horizontal-left-spacer"
                    ref="eLeftSpacer"
                    style="width: 0px; max-width: 0px; min-width: 0px;"
                  />
                  
            
                  <div
                    class="ag-body-horizontal-scroll-viewport"
                    ref="eViewport"
                    style="height: 0px; max-height: 0px; min-height: 0px;"
                  >
                    
                
                    <div
                      class="ag-body-horizontal-scroll-container"
                      ref="eContainer"
                      style="height: 0px; max-height: 0px; min-height: 0px; width: 1230px;"
                    />
                    
            
                  </div>
                  
            
                  <div
                    class="ag-horizontal-right-spacer"
                    ref="eRightSpacer"
                    style="width: 0px; max-width: 0px; min-width: 0px;"
                  />
                  
        
                </div>
                <!-- AG Overlay Wrapper -->
                <div
                  aria-hidden="true"
                  class="ag-overlay"
                >
                  
            
                  <div
                    class="ag-overlay-panel"
                  >
                    
                
                    <div
                      class="ag-overlay-wrapper ag-layout-auto-height ag-overlay-no-rows-wrapper"
                      ref="eOverlayWrapper"
                    >
                      <span
                        class="ag-overlay-no-rows-center"
                      >
                        No Rows To Show
                      </span>
                    </div>
                    
            
                  </div>
                  
        
                </div>
              </div>
              <div
                class="ag-tab-guard ag-tab-guard-bottom"
                role="presentation"
                tabindex="0"
              />
            </div>
            <div
              class="ag-paging-panel ag-unselectable ag-hidden"
              id="ag-15"
            >
              
                
              <span
                class="ag-paging-row-summary-panel"
                role="status"
              >
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-15-first-row"
                  ref="lbFirstRowOnPage"
                />
                
                    
                <span
                  id="ag-15-to"
                >
                  to
                </span>
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-15-last-row"
                  ref="lbLastRowOnPage"
                />
                
                    
                <span
                  id="ag-15-of"
                >
                  of
                </span>
                
                    
                <span
                  class="ag-paging-row-summary-panel-number"
                  id="ag-15-row-count"
                  ref="lbRecordCount"
                />
                
                
              </span>
              
                
              <span
                class="ag-paging-page-summary-panel"
                role="presentation"
              >
                
                    
                <div
                  aria-label="First Page"
                  class="ag-paging-button"
                  ref="btFirst"
                  role="button"
                >
                  <span
                    class="ag-icon ag-icon-first"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <div
                  aria-label="Previous Page"
                  class="ag-paging-button"
                  ref="btPrevious"
                  role="button"
                >
                  <span
                    class="ag-icon ag-icon-previous"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <span
                  class="ag-paging-description"
                  role="status"
                >
                  
                        
                  <span
                    id="ag-15-start-page"
                  >
                    Page
                  </span>
                  
                        
                  <span
                    class="ag-paging-number"
                    id="ag-15-start-page-number"
                    ref="lbCurrent"
                  />
                  
                        
                  <span
                    id="ag-15-of-page"
                  >
                    of
                  </span>
                  
                        
                  <span
                    class="ag-paging-number"
                    id="ag-15-of-page-number"
                    ref="lbTotal"
                  />
                  
                    
                </span>
                
                    
                <div
                  aria-label="Next Page"
                  class="ag-paging-button"
                  ref="btNext"
                  role="button"
                >
                  <span
                    class="ag-icon ag-icon-next"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                    
                <div
                  aria-label="Last Page"
                  class="ag-paging-button"
                  ref="btLast"
                  role="button"
                >
                  <span
                    class="ag-icon ag-icon-last"
                    role="presentation"
                    unselectable="on"
                  />
                </div>
                
                
              </span>
              
            
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="button"
      >
        Add to cart
      </button>
    </div>
  </div>
</DocumentFragment>
`;
